Marlin/MarlinSerial.cpp

```diff
@@ -325,144 +325,72 @@ MarlinSerial customizedSerial;
 
     enum e_parser_state {
       state_RESET,
+      state_N,
       state_M,
       state_M1,
       state_M10,
       state_M11,
-      state_M2,
-      state_M3,
       state_M4,
       state_M41,
       state_IGNORE // to '\n'
     };
 
     static e_parser_state state = state_RESET;
 
+    if (c == '\n') state = state_IGNORE;
+
     switch (state) {
       case state_RESET:
         switch (c) {
-          case 'M':
-            state = state_M;
-            break;
-          case ';':
-            state = state_IGNORE;
-            break;
-          default: state = state_RESET;
+          case ' ': break;
+          case 'N': state = state_N;      break;
+          case 'M': state = state_M;      break;
+          default: state = state_IGNORE;
         }
-      break;
+        break;
+
+      case state_N:
+        if (!NUMERIC_SIGNED(c)) state = state_RESET;
+        break;
```

Comment:
I can't see what the state_N is for. 
Just ignore everything else than a 'M' in state_RESET and stay in state_RESET.